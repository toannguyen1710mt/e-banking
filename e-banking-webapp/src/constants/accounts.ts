// Interfaces
import { AccountType } from '@/interfaces';

export const ACCOUNT_TYPES = {
  MAIN_WALLET: 'Main Wallet',
  SAVINGS: 'Savings',
  CHECKING: 'Checking',
};

export const TRANSFER_FORM_ACCOUNT_OPTIONS = [
  {
    key: AccountType.MAIN,
    label: 'Main Account',
  },
  {
    key: AccountType.SAVINGS,
    label: 'Savings',
  },
  {
    key: AccountType.CHECKING,
    label: 'Checking',
  },
];
