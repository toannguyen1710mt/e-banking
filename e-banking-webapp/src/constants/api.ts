export const API_ENDPOINTS = {
  USERS: '/users',
  ACCOUNTS: '/accounts',
  GLOBAL_ACCOUNTS: '/global-accounts',
  TRANSACTIONS: '/transactions',
  CARDS: '/cards',
  SIGN_UP: '/auth/local/register',
  CHANGE_PASSWORD: '/auth/change-password',
};

export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;

export const END_POINT = {
  UPLOAD: `/upload`,
};
