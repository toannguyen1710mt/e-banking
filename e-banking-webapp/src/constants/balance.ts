import { CardStatistics } from '@/interfaces';

export const EXPENSE_LABEL_DATA = ['Amazon', 'Netflix', 'Apple', 'Figma'];
export const EXPENSE_VALUE_DATA = ['#8AB17D', '#2A9D8F', '#F4A261', '#E9C46A'];
export const EXPENSE_COLOR_CLASSES = [
  'bg-[#8AB17D]',
  'bg-[#2A9D8F]',
  'bg-[#F4A261]',
  'bg-[#E9C46A]',
];
export const AVAILABLE_WALLETS = [
  CardStatistics.Savings,
  CardStatistics.Checking,
];
export const CARD_STATISTICS = [
  {
    label: CardStatistics.Balance,
    color: 'bg-java',
  },
  {
    label: CardStatistics.Savings,
    color: 'bg-pastelYellow',
  },
  {
    label: CardStatistics.Checking,
    color: 'bg-softGreen',
  },
];
