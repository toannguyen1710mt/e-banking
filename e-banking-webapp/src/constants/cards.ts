export const MODAL_STYLES_ADD_CREDIT_CARD = {
  true: {
    cardMaxWidth: 'max-w-[195px]',
    cardHeaderGap: 'gap-0',
    headerContentGap: 'gap-1',
    payPassIcon: 'w-[11px] h-[14px]',
    cardBody: 'mb-[10px] mt-[8px]',
    chipSize: 'w-[24px] h-[16px]',
    worldTextSize: 'text-5xs',
    cardNumberText: 'ml-2 mt-0 text-xs',
    detailsGap: 'mt-0 gap-2',
    expireText: 'text-3xs',
    holderNameText: 'max-w-[115px] text-3xs',
    mastercardIconSize: 'w-[35px]',
  },
  false: {
    cardMaxWidth: 'max-w-[290px]',
    cardHeaderGap: 'gap-2',
    headerContentGap: 'gap-3',
    payPassIcon: 'unset',
    cardBody: 'mb-[18px] mt-[14px]',
    chipSize: 'unset',
    worldTextSize: 'text-2xs',
    cardNumberText: 'ml-4 mt-1 text-sm',
    detailsGap: 'mt-2 gap-5',
    expireText: 'text-2xs',
    holderNameText: 'max-w-[185px] text-xs',
    mastercardIconSize: 'unset',
  },
};
