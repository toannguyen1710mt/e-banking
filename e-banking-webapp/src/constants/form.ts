// Interfaces
import { AccountType, CurrencyUnit } from '@/interfaces';

export const SIGNUP_FORM_DEFAULT_VALUES = {
  user: {
    username: '',
    email: '',
    password: '',
    confirmPassword: '',
  },
  contact: {
    phone: '',
    country: 'United States',
    postal: '344',
  },
  card: {
    holderName: '',
    cardNumber: '',
    expireAt: '',
    ccv: '',
  },
};

export const ACCOUNT_DEFAULT_VALUES = [
  {
    accountNumber: '123456789208',
    balance: 3438800,
    type: AccountType.MAIN,
    currency: '$' as CurrencyUnit,
    name: 'OCB',
  },
  {
    accountNumber: '999332289888',
    balance: 84438800,
    type: AccountType.CHECKING,
    currency: '$' as CurrencyUnit,
    name: 'ACB',
  },
  {
    accountNumber: '106868929357',
    balance: 9238800,
    type: AccountType.SAVINGS,
    currency: '$' as CurrencyUnit,
    name: 'VID',
  },
];

export const PASSWORD_DEFAULT_VALUES = {
  currentPassword: '',
  newPassword: '',
  confirmPassword: '',
};
