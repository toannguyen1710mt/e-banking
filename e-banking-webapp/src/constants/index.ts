export * from './accounts';
export * from './routes';
export * from './themes';
export * from './messages';
export * from './images';
export * from './regex';
export * from './navbar';
export * from './titles';
export * from './global-transfer';
export * from './balance';
export * from './charts';
export * from './form';
export * from './api';
export * from './tags';
export * from './settings';
export * from './tables';
export * from './cards';
export * from './date';
export * from './sidebar-list';
