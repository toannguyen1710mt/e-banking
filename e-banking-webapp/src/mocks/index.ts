export * from './options';
export * from './card';
export * from './charts';
export * from './transaction';
export * from './table';
export * from './transferConfirmData';
export * from './target';
export * from './analytics';
export * from './auth';
export * from './ServiceCard';
export * from './formMocks';
