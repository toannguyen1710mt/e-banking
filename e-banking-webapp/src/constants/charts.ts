import { ApexOptions } from 'apexcharts';
import { EXPENSE_LABEL_DATA, EXPENSE_VALUE_DATA } from './balance';

export const createExpenseAnalysisOptions = (value: string): ApexOptions => ({
  chart: {
    type: 'donut',
  },
  labels: EXPENSE_LABEL_DATA,
  responsive: [
    {
      breakpoint: 320,
      options: {
        chart: {
          width: '100%',
        },
        legend: {
          position: 'bottom',
        },
      },
    },
  ],
  legend: {
    show: false,
  },
  dataLabels: {
    enabled: false,
  },
  plotOptions: {
    pie: {
      donut: {
        size: '70%',
        labels: {
          show: true,
          name: {
            show: false,
          },
          value: {
            show: true,
            fontSize: '8px',
            fontWeight: 800,
            color: '#fff',
            offsetY: 3,
          },
          total: {
            show: true,
            formatter: () => {
              const formattedValue = `${value}`;

              return formattedValue.length > 7
                ? `${formattedValue.slice(0, 5)}...`
                : formattedValue;
            },
          },
        },
      },
    },
  },
  stroke: {
    width: 1,
  },

  colors: EXPENSE_VALUE_DATA,
});
