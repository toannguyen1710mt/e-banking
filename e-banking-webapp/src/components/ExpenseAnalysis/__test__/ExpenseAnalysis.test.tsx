import { render } from '@testing-library/react';

// Constants
import { createExpenseAnalysisOptions } from '@/constants';

// Mocks data
import { MOCK_SERIES_EXPENSE_ANALYSIS } from '@/mocks';

// Components
import { ExpenseAnalysis } from '..';

jest.mock('react-apexcharts', () => {
  return {
    __esModule: true,
    default: () => <div>Mock ApexCharts</div>,
  };
});

jest.mock('@/constants/rules', () => ({
  createStepSchema: jest.fn(),
}));

describe('ExpenseAnalysis component', () => {
  it('Should render snapshot correctly', () => {
    expect(
      render(
        <ExpenseAnalysis
          series={MOCK_SERIES_EXPENSE_ANALYSIS}
          options={createExpenseAnalysisOptions('220 000')}
        />,
      ),
    ).toMatchSnapshot();
  });
});
