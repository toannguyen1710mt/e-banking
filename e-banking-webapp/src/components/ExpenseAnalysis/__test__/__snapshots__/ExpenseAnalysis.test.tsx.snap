// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseAnalysis component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none max-w-[250px] bg-background-300 p-[10px]"
        tabindex="-1"
      >
        <div
          class="flex z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large p-0"
        >
          <p
            class="text-foreground-200 text-base font-medium"
          >
            Expense Analysis
          </p>
        </div>
        <div
          class="mt-2 flex gap-4"
        >
          <div
            class="grid grid-cols-2 content-center gap-x-5 gap-y-3"
          >
            <div
              class="flex h-3 items-center gap-[5px]"
            >
              <div
                class="h-[8px] w-[8px] rounded-full bg-[#8AB17D]"
              />
              <p
                class="text-primary-200 font-medium text-2xs text-white opacity-50"
              >
                Amazon
              </p>
            </div>
            <div
              class="flex h-3 items-center gap-[5px]"
            >
              <div
                class="h-[8px] w-[8px] rounded-full bg-[#2A9D8F]"
              />
              <p
                class="text-primary-200 font-medium text-2xs text-white opacity-50"
              >
                Netflix
              </p>
            </div>
            <div
              class="flex h-3 items-center gap-[5px]"
            >
              <div
                class="h-[8px] w-[8px] rounded-full bg-[#F4A261]"
              />
              <p
                class="text-primary-200 font-medium text-2xs text-white opacity-50"
              >
                Apple
              </p>
            </div>
            <div
              class="flex h-3 items-center gap-[5px]"
            >
              <div
                class="h-[8px] w-[8px] rounded-full bg-[#E9C46A]"
              />
              <p
                class="text-primary-200 font-medium text-2xs text-white opacity-50"
              >
                Figma
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none max-w-[250px] bg-background-300 p-[10px]"
      tabindex="-1"
    >
      <div
        class="flex z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large p-0"
      >
        <p
          class="text-foreground-200 text-base font-medium"
        >
          Expense Analysis
        </p>
      </div>
      <div
        class="mt-2 flex gap-4"
      >
        <div
          class="grid grid-cols-2 content-center gap-x-5 gap-y-3"
        >
          <div
            class="flex h-3 items-center gap-[5px]"
          >
            <div
              class="h-[8px] w-[8px] rounded-full bg-[#8AB17D]"
            />
            <p
              class="text-primary-200 font-medium text-2xs text-white opacity-50"
            >
              Amazon
            </p>
          </div>
          <div
            class="flex h-3 items-center gap-[5px]"
          >
            <div
              class="h-[8px] w-[8px] rounded-full bg-[#2A9D8F]"
            />
            <p
              class="text-primary-200 font-medium text-2xs text-white opacity-50"
            >
              Netflix
            </p>
          </div>
          <div
            class="flex h-3 items-center gap-[5px]"
          >
            <div
              class="h-[8px] w-[8px] rounded-full bg-[#F4A261]"
            />
            <p
              class="text-primary-200 font-medium text-2xs text-white opacity-50"
            >
              Apple
            </p>
          </div>
          <div
            class="flex h-3 items-center gap-[5px]"
          >
            <div
              class="h-[8px] w-[8px] rounded-full bg-[#E9C46A]"
            />
            <p
              class="text-primary-200 font-medium text-2xs text-white opacity-50"
            >
              Figma
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
